util/gogc.go

```diff
@@ -40,10 +40,62 @@ func SetGOGC(val int) {
 	}
 	debug.SetGCPercent(val)
 	metrics.GOGC.Set(float64(val))
-	atomic.StoreInt64(&gogcValue, int64(val))
+	gogcValue.Store(int64(val))
 }
 
 // GetGOGC returns the current value of GOGC.
 func GetGOGC() int {
-	return int(atomic.LoadInt64(&gogcValue))
+	return int(gogcValue.Load())
+}
+
+// GOGCStatSampler is a sampler for GOGC.
+type GOGCStatSampler struct {
+	last struct {
+		now         int64
+		lastGC      time.Time
+		gcPauseTime uint64
```

Comment:
Rename `gcPauseTime` to `gcPauseTotal` would be better?

Reply:
Done